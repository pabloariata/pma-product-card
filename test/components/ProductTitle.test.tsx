import React from 'react';
import  renderer  from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';



describe('Pruebas en ProductTitle', () => { 
   
    test('Debe de mostrar el componente con el titulo personalizado', () => { 
        

        const wrapper = renderer.create(
            <ProductTitle title="Titulo Custom" className='clase-custom'/>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

    test('Debe de mostrar el titulo por defecto si no se especifica un titulo', () => { 
       
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();


    });

});