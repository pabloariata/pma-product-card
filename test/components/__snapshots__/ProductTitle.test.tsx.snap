// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductTitle Debe de mostrar el componente con el titulo personalizado 1`] = `
<span
  className="productDescription clase-custom"
>
  Titulo Custom
</span>
`;

exports[`Pruebas en ProductTitle Debe de mostrar el titulo por defecto si no se especifica un titulo 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug!
  </span>
</div>
`;
